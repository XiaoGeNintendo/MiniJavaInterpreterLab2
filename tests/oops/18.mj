class A{
    int x=1;
    void foo(){
        print(this.x);
        this.shout();
    }
    void ask(){}
    void shout(){print("A");}
}

class B extends A{
    int x=2;
    void foo(){print(this.x);this.shout();}
    void ask(){super.foo();}
    void shout(){print("B");}
}

int main(){
    A x=new B();
    x.ask();
    println();
    return 0;
}