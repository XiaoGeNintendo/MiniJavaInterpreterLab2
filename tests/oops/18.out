1B
Process exits with the code 0.